src/site/antora/modules/ROOT/pages/manual/filters.adoc

```diff
@@ -17,1038 +17,1407 @@
-      <PatternLayout>
-        <pattern>%d %p %c{1.} [%t] %m%n</pattern>
-      </PatternLayout>
-    </File>
-    <RollingFile name="RollingFile" fileName="logs/app.log"
-                 filePattern="logs/app-%d{MM-dd-yyyy}.log.gz">
-      <BurstFilter level="INFO" rate="16" maxBurst="100"/>
-      <PatternLayout>
-        <pattern>%d %p %c{1.} [%t] %m%n</pattern>
-      </PatternLayout>
-      <TimeBasedTriggeringPolicy />
-    </RollingFile>
-  </Appenders>
-  <Loggers>
-    <Logger name="EventLogger" level="info">
-      <AppenderRef ref="Audit"/>
-    </Logger>
-    <Root level="error">
-      <AppenderRef ref="RollingFile"/>
-    </Root>
-  </Loggers>
-</Configuration>
+include::example$manual/filters/TimeFilter.properties[tag=filter]
 ----
+====
 
-[#DynamicThresholdFilter]
-=== DynamicThresholdFilter
+xref:plugin-reference.adoc#org-apache-logging-log4j_log4j-core_org-apache-logging-log4j-core-filter-TimeFilter[📖 Plugin reference for `TimeFilter`]
 
-The DynamicThresholdFilter allows filtering by log level based on specific attributes.
-For example, if the user's loginId is being captured in the ThreadContext Map then it is possible to enable debug logging for only that user.
-If the log event does not contain the specified ThreadContext item NEUTRAL will be returned.
+[#level-filters]
+=== Level filters
 
-.Dynamic Threshold Filter Parameters
-[cols="1m,1,4"]
+The following filters allow you to filter log events based on their level.
+
+[#LevelMatchFilter]
+==== `LevelMatchFilter`
+
+The `LevelMatchFilter` matches log events that have exactly a certain log level.
+
+Besides the common configuration attributes, the `LevelMatchFilter` supports the following parameter:
+
+.`LevelMatchFilter` -- configuration attributes
+[cols="1m,1,1,4"]
 |===
-|Parameter Name |Type |Description
-
-|key
-|String
-|The name of the item in the ThreadContext Map to compare.
-
-|defaultThreshold
-|String
-|Level of messages to be filtered. The default
-threshold only applies if the log event contains the specified
-ThreadContext Map item and its value do not match any key in the
-key/value pairs.
-
-|keyValuePair
-|KeyValuePair[]
-|One or more KeyValuePair elements that
-define the matching value for the key and the Level to evaluate when the
-key matches.
-
-|onMatch
-|String
-|Action to take when the filter matches. Can be ACCEPT,
-DENY or NEUTRAL. The default value is NEUTRAL.
-
-|onMismatch
-|String
-|Action to take when the filter does not match. Can
-be ACCEPT, DENY or NEUTRAL. The default value is DENY.
+| Attribute | Type | Default value | Description
+
+| level
+| link:../javadoc/log4j-api/org/apache/logging/log4j/Level.html[`Level`]
+| link:../javadoc/log4j-api/org/apache/logging/log4j/Level.html#ERROR[`ERROR`]
+| The filter only matches log events of this level.
+
 |===
 
-Here is a sample configuration containing the DynamicThresholdFilter:
+[TIP]
+====
+If you wish to use a different log file for each log level, you can also use a
```

Comment:
I think they should.

Personally I don't find the practice of splitting each log level in a separate file very useful, since you need to look at several files to give a context to each event.
However, in UNIX the `/var/log/mail.*.log` files have been traditionally split per level, so we want to give users the same opportunity here.

Fixed in 4d0d7aa6c4c2ca8f9ad92621e50f8b513e59ec71.

Reply:
Please note that my question wasn't `can or should they split logs by level`, but `can or should they use routing appender instead`.

Reply:
Yes, I understood.

My rant was about splitting logs by level, but my fix explains why they should use `Routing` instead of `LevelMatchFilter`. :wink: